src/openssl.c

```diff
@@ -3249,17 +3249,49 @@ static BIO *getbio(lua_State *L) {
 	bio = lua_touserdata(L, -1);
 	lua_pop(L, 1);
 
-	BIO_reset(*bio);
-
 	return *bio;
+} /* getbioinstance() */
+
+
+static BIO *getbio(lua_State *L) {
+	BIO *bio = getbioinstance(L);
+	BIO_reset(bio);
+	return bio;
 } /* getbio() */
 
 
+static void pushbiostring(lua_State *L) {
+	BIO *bio = getbioinstance(L);
+	char *str;
+	long len = BIO_get_mem_data(bio, &str);
+	lua_pushlstring(L, str, len);
+	BIO_reset(bio);
+} /* pushbiostring() */
+
+
+static int pem_pw_cb(char *buf, int size, int rwflag, void *u) {
+	lua_State *L = (lua_State *) u;
+
+	if (lua_isnil(L, -1))
+		return 0;
+
+	if (lua_isfunction(L, -1))
```

Comment:
A type check should be *before* we get to the callback.

Reply:
What would be the benefit? The current callback supports multiple types, so a check is needed there anyway.